### PR TITLE
Maint/pre commit

### PR DESCRIPTION
**Pull request by justinGilmer**
As is the case with foyer and mBuild, supporting autoformatting and various lint checks before committing makes reviewing pull requests much simpler for reviewers and contributors.

This PR is the initial setup for pre-commit in gmso.

However, one thing to note is that the documentation we get from pydantic does not 100% align with the numpydoc style that `pydocstyle` checks, which led to a lot of difficulty initially. @umesh-timalsina and I decided that we did not want to lose the useful features of the pydantic documentation we currently have set up. So a few of the main pydantic-heavy objects we use in gmso `gmso/core, gmso/abc` are excluded from the pydocstyle check. I added quite a few docstrings to these objects anyway, but it should not be required.

### CODE REVIEW

File: gmso/core/improper.py [hunk 1]
```diff
@@ -31,80 +32,89 @@ class Improper(Connection):
     """
     connection_members_: Tuple[Atom, Atom, Atom, Atom] = Field(
         ...,
-        description='The 4 atoms of this improper. Central site first, '
-                    'then the three atoms connected to the central site.'
+        description="The 4 atoms of this improper. Central site first, "
```
Comment (umesh-timalsina) [inline gmso/core/improper.py:35]:
```suggestion
        description="The 4 atoms of this improper. Central atom first, "
```

**Review by daico007 — APPROVED** [review]
LGTM!

**Comment by umesh-timalsina** [comment]
pre-commit.ci run

**Review by umesh-timalsina — APPROVED** [review]
LGTM!!

**Pull request merged**